Gordon360/Services/EventService.cs

```diff
@@ -27,223 +24,52 @@ namespace Gordon360.Services
         }
 
-
         /// <summary>
-        /// Return a Single Event JObject from Live25
+        /// Select only events that are marked for Public promotion
         /// </summary>
-        /// <returns></returns>
-        public IEnumerable<EventViewModel> GetSpecificEvents(string EventID, string type)
+        /// <returns>All Public Events</returns>
+        public IEnumerable<EventViewModel> GetPublicEvents()
         {
-
-            // Get API Route for 25 Live
-            string requestUrl = GetRoute(EventID, type);
-            // Commit contents of the request to temporary memory
-            XDocument xmlDoc = Helpers.GetLiveStream(requestUrl);
-
-            // Pull out the nodes for events
-            IEnumerable<XElement> events = xmlDoc.Descendants(r25 +"event");
-
-            // Convert to iterable list containing just the pieces we need
-            List<EventViewModel> stuff = new List<EventViewModel>();
-
-            // Convert each element into a viewmodel for events          
-            foreach (XElement n in events)
-            {
-                EventViewModel vm = new EventViewModel(n);
-                stuff.Add(vm);
-            }
-
-            return stuff.AsEnumerable<EventViewModel>();
-            
+            return GetAllEvents().Where(e => e.IsPublic);
         }
-        
+
         /// <summary>
-        /// Access the memory stream created by the cached task and parse it into events
+        /// Select only events that are Approved to give CLAW credit
         /// </summary>
-        /// <param name="xmlDoc"></param>
-        /// <returns></returns>
-        public IEnumerable<EventViewModel> GetAllEvents (XDocument xmlDoc)
+        /// <returns>All CLAW Events</returns>
+        public IEnumerable<EventViewModel> GetCLAWEvents()
         {
-
-                // Pull out the nodes for events
-                IEnumerable<XElement> events = xmlDoc.Descendants(r25 + "event");
-
-                // Convert to iterable list containing just the pieces we need
-                List<EventViewModel> stuff = new List<EventViewModel>();
-
-                // Convert each element into a viewmodel for events          
-                foreach (XElement n in events)
-                {
-                    EventViewModel vm = new EventViewModel(n);
-                    stuff.Add(vm);
-                }
-
-                return stuff.AsEnumerable<EventViewModel>();
-            
+            return GetAllEvents().Where(e => e.HasCLAWCredit);
         }
 
+
         /// <summary>
-        /// Returns all attended events for a student
+        /// Returns all attended events for a student in a specific term
         /// </summary>
         /// <param name="user_name"> The student's ID</param>
+        /// <param name="term"> The current term</param>
         /// <returns></returns>
         [StateYourBusiness(operation = Operation.READ_PARTIAL, resource = Resource.ChapelEvent)]
```

Comment:
The authorization filter has been moved, although I'm not sure it does anything since we don't have any restrictions on the chapelEvent resource for Partial read actions. This might be something to investigate when we check on route security.